units/host_monitoring_idle_termination.go

```diff
@@ -263,12 +269,15 @@ func (i hostIdleInfo) getTerminationReason() string {
 		// busy.
 		return "host has an outdated AMI"
 	}
-	if i.timeSinceLastCommunication >= i.idleThreshold {
+	if i.timeSinceLastCommunication >= i.idleThreshold && !i.isRunningTearDownTaskGroup {
 		return fmt.Sprintf("host is idle or unreachable, communication time %s is over threshold time %s", i.timeSinceLastCommunication, i.idleThreshold)
 	}
-	if i.idleTime >= i.idleThreshold {
+	if i.idleTime > 0 && i.idleTime >= i.idleThreshold {
 		return fmt.Sprintf("host is idle or unreachable, idle time %s is over threshold time %s", i.idleTime, i.idleThreshold)
 	}
+	if i.teardownTimeExceededMax {
```

Comment:
Not a nit, but should we combine `timeSinceTaskGroupTeardownStartTime` and `teardownTimeExceededMax`? It seems like `teardownTimeExceededMax` is just the first one checked against a constant. I'm fine with keeping them as separate but seems like extra variables/names since this bool is only used here